src/ec/system76/ec/acpi/thermal.asl

```diff
@@ -0,0 +1,162 @@
+Scope (\_TZ)
+{
+	Device (TFN1)
+	{
+		Name (_HID, EISAID ("PNP0C0B"))
+		Name (_UID, 0)
+		Name (_STR, Unicode("CPU Fan"))
+
+		/* _FIF: Fan Information */
+		Name (_FIF, Package ()
+		{
+			0,	// Revision
+			1,	// Fine Grained Control
+			1,	// Step Size
+			0	// No Low Speed Notification
+		})
+
+		Name (TFST, Package ()
+		{
+			0,	// Revision
+			0x00,	// Control
+			0x00	// Speed
+		})
+
+		/* _FST: Fan current Status */
+		Method (_FST, 0, Serialized,,PkgObj)
```

Comment:
I think only fan status is needed here. All Fan methods are optional according to ACPI spec.

Reply:
ACPI spec confusingly also [states](https://uefi.org/htmlspecs/ACPI_Spec_6_4_html/11_Thermal_Management/fan-device.html): `While the Fan Device and its associated objects are optional, if the Fan Device is implemented by the platform, all objects listed in the table above are required and must be provided.`

Reply:
Ahh okay...

Reply:
And linux also expects all 4 to be implemented: https://github.com/torvalds/linux/blob/7dd4b804e08041ff56c88bdd8da742d14b17ed25/drivers/acpi/fan_core.c#L206-L212